### ISSUE TITLE
Simplify comparing LQPs for equality

### ISSUE THREAD

**Issue by mrzzzrm** [hyrise/hyrise#612]
I take issue with the way we're testing LQP structure. Look at `SQLTranslatorTest` or `JoinDetectionRuleTest`. Both try to verify the structure of LQPs by painstakingly checking individual fields in nodes. They do this in a very cumbersome and error prone way.

`TEST_F(SQLTranslatorTest, TwoColumnFilter)` tests a super simple query (`SELECT * FROM table_a WHERE a = "b"`), but already has seven `EXPECT/ASSERT`s in it! Yet it already fail to check multiple things (Are actually all columns of table_a selected? Does the projection node even have a left child? Does the predicate node have a StoredTableNode as left child?)! 

**It would be much simpler (to read and to write) and more thorough, for all of these tests to just build the expected LQP and compare it to the generated LQP using something like `EXPECT_LQP_EQUAL`.  Upon failure, this macro could just print the LQPs** making the error much more obvious than the current endless cascade of EXPECTs that we have right now.
When touching all of these Tests when implementing the LQPColumnOrigins I stumbled upon countless points where these tests were incomplete or insecure (not checking for node types before casting, etc.) It being so cumbersome to test LQP structure also discourages people to write tests in the first place (bad, bad, bad!!)

Long story short: I want `EXPECT_LQP_EQUAL`.
